.github/workflows/test_common.yml

```diff
@@ -118,6 +118,9 @@ jobs:
       - name: Install pyarrow
         run: poetry install --no-interaction -E duckdb -E cli -E parquet --with sentry-sdk
 
+      - name: Ensure pyarrow 15
+        run: poetry run pip install pyarrow==15.0.2
+
```

Comment:
OK! just make sure that this line:
```
poetry install --no-interaction -E duckdb -E cli -E parquet -E deltalake -E sql_database --with sentry-sdk,pipeline,sources
```
is not installing arrow 17 again. I see delta  is being installed so probably that happens... maybe we should split testing of the `lib` to a separate step where we install delta

Reply:
yeah good point. I think every call to poetry install has the potential to change this. I have updated the workflow not fix this, but I am not quite sure how you would set it up tbh.